modules/core/src/main/scala/org/scalasteward/core/coursier/CoursierAlg.scala

```diff
@@ -50,64 +42,72 @@ object CoursierAlg {
       parallel: Parallel[F],
       F: Async[F]
   ): CoursierAlg[F] = {
-    val fetch: Fetch[F] = Fetch[F](FileCache[F]())
+    val fetch: Fetch[F] =
+      Fetch[F](FileCache[F]())
 
     val cacheNoTtl: FileCache[F] =
       FileCache[F]().withTtl(None).withCachePolicies(List(CachePolicy.Update))
 
     new CoursierAlg[F] {
-      override def getArtifactUrl(dependency: Scope.Dependency): F[Option[Uri]] =
-        convertToCoursierTypes(dependency).flatMap((getArtifactUrlImpl _).tupled)
+      override def getMetadata(
+          dependency: Dependency,
+          resolvers: List[Resolver]
+      ): F[Option[DependencyMetadata]] =
+        resolvers.traverseFilter(convertResolver(_).attempt.map(_.toOption)).flatMap {
+          repositories =>
+            val csrDependency = toCoursierDependency(dependency)
+            getMetadataImpl(csrDependency, repositories, None)
+        }
 
-      private def getArtifactUrlImpl(
+      private def getMetadataImpl(
           dependency: coursier.Dependency,
-          repositories: List[coursier.Repository]
-      ): F[Option[Uri]] = {
+          repositories: List[coursier.Repository],
+          acc: Option[DependencyMetadata]
+      ): F[Option[DependencyMetadata]] = {
```

Comment:
just an idea:
you could simplify the `acc` to `acc: DependencyMetadata = DependencyMetadata(None,None,None)`
And the return type would be `F[DependencyMetadata]`

The further down the `fold` gets simpler, too:
instead of 
```Scala
val metadata = {
    val current = metadataFrom(project)
    acc.fold(current)(_.enrichWith(current))
}
```

you could write:
```Scala
val metadata = acc.enrichWith(metadataFrom(project))
```

Reply:
That is a good suggestion. We currently do not care about the difference of an empty `Option[DependencyMetadata]` and an empty `DependencyMetadata`. And dealing with the latter is easier in a few places.

Reply:
Done in [7e0bec6](https://github.com/scala-steward-org/scala-steward/pull/2863/commits/7e0bec6a3cb42f5af754f6a1b2e9ceb1e93c9464).